### PR TITLE
Add arcade game Frogger

### PR DESCRIPTION
**Pull request by Vitaminvp**
### Arcade game Frogger

[Demo](https://vitaminvp.github.io/kottans-frontend/task_Frogger/) |
[Code base](https://github.com/Vitaminvp/kottans-frontend/tree/master/task_Frogger)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

**Comment by OleksiyRudenko** [comment]
@Vitaminvp Looks nice! Please, check yourself against the following list:

 * [ ] it is OK to employ ES6 features like `const`, `let` etc.
 * [ ] OO is implemented using JS prototype chain object model (**not** ES6 classes syntax)
 * Requirements re **Constants**:
   * [ ] all numbers like block dimensions, initial locations are defined as constants
   * [ ] there are core contants and derived constants
         (e.g. `const FIELD_WIDTH = BLOCK_WIDTH * BLOCKS_NUMBER;`)
   * [ ] arrays of constants are also constants 
         (e.g. `const INITIAL_POSITIONS = [1,2,3,4].map(rowNumber => rowNumber * BLOCK_HEIGHT);`)
   * [ ] const objects help organizing and structure const data even better
         (e.g. `const PLAYER_CONF = { initialPosition: {x: 1, y: 5}, sprite: '...', ...etc... };`
 * Requirements re **OOP**:
   * [ ] properties common for some classes are generalized into a base class
         (e.g. there is `Character` base class, which is extended by `Enemy` and `Player` classes)
   * [ ] class extension is implemented using `Subclass.prototype = Object.create(Superclass.prototype)`,
         not `Subclass.prototype = new Superclass(params);`;
         [Useful resource](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Inheritance_and_the_prototype_chain)
   * [ ] classes do not refer to any global variables, like global variable `player`, which is an instance of `Player` class
         (referring to global constants and globals provided by the gaming platform like `Resources` is OK);
         Hint: pass `Player` instance as an argument to every enemy
 * Most common mistakes
   * [ ] Make sure `target = condition ? valueWhenConditionTrue : valueWhenConditionFalse` is used instead of
         `condition ? target = valueWhenConditionTrue : target = valueWhenConditionFalse`;
         [Conditional (ternary) operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator)
         
Please, confirm all items above are positive. Please, ask a question if anything above requires clarification.

**Comment by Vitaminvp** [comment]
Thanks. I tried to fix code according to remarks mentioned above.

File: submissions/Vitaminvp/js-oop-frogger/app.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+var Player = function(x, y) {
+    this.player = SPRITES[0];//"images/char-boy.png";
+    this.x = x;
+    this.y = y;
+};
+
+Player.prototype.update = function() {};
+
+Player.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.player), this.x, this.y);
```
Comment (OleksiyRudenko) [inline submissions/Vitaminvp/js-oop-frogger/app.js]: Notice how similar are `render` methods from `Player` and `Enemy`. This is a good sign of the fact that both characters classes can benefit from having a base class, so you once you decide to change rendering approach you do this in a single place.
Imagine that you also have gems, health items, different types of enemies etc.

You are not required to fix this though. Just keep in mind.

**Comment by Vitaminvp** [comment]
Thanks. Could you please look again.

File: submissions/Vitaminvp/js-oop-frogger/app.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    // You should multiply any movement by the dt parameter
+    // which will ensure the game runs at the same speed for
+    // all computers.
+    this.x += this.speed * dt;
+
+    handleCollision(this.player, this);
+
+};
+
+function handleCollision(player, enemy) {
```
Comment (OleksiyRudenko) [inline submissions/Vitaminvp/js-oop-frogger/app.js]: > Please, put collision handling into a dedicated method.

The difference between method and function is that method is a function that belongs to a class.

Please, fix.

**Comment by OleksiyRudenko** [comment]
Please, keep in mind that resolving conversations you have elaborated on is a good way to let reviewer know you did.

**Comment by Vitaminvp** [comment]
How about that?

**Comment by Vitaminvp** [comment]
Thanks :-)

**Comment by Vitaminvp** [comment]
I really appreciate your attention to details. Thanks again.

**Pull request merged**